dev/io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat/fat/src/io/openliberty/microprofile/telemetry/internal/utils/otelCollector/OtelCollectorContainer.java

```diff
@@ -0,0 +1,76 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.microprofile.telemetry.internal.utils.otelCollector;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+import org.testcontainers.utility.ImageNameSubstitutor;
+
+import java.io.File;
+/**
+ * A container for the otelCollector trace server
+ * <p>
+ * Usage:
+ *
+ * <pre>
+ * {@code @ClassName}
+ * public static otelCollectorContainer otelCollectorContainer = new OtelCollectorContainer().withLogConsumer(new SimpleLogConsumer(MyTest.class, "jaeger"));
+ * </pre>
+ *
+ */
+
+public class OtelCollectorContainer extends GenericContainer<OtelCollectorContainer> {
+
+    public static final DockerImageName IMAGE_NAME = DockerImageName.parse("otel/opentelemetry-collector:0.74.0");
+
+    public static final int OTLP_GRPC_PORT = 4317;
+
+    public OtelCollectorContainer(String exporter) {
+        this(IMAGE_NAME, exporter);
+    }
+
+    public OtelCollectorContainer(DockerImageName imageName, String exporter) {
+        super(new ImageFromDockerfile().withDockerfileFromBuilder(builder -> builder.from(
+                                                                        ImageNameSubstitutor.instance()
+                                                                        .apply(DockerImageName.parse("otel/opentelemetry-collector:0.74.0"))
+                                                                        .asCanonicalNameString())
+                                                                        .copy("/etc/otel-collector-config.yaml", "/etc/otel-collector-config.yaml")
+                                                                        .build())
+                                                                        .withFileFromFile("/etc/otel-collector-config.yaml", new File("lib/LibertyFATTestFiles/otel-collector-config-"+exporter+".yaml"), 0644));
+        withExposedPorts(OTLP_GRPC_PORT);
+        withCommand("--config=/etc/otel-collector-config.yaml");
+    }
```

Comment:
Pass the `File` object as a parameter to the constructor rather than building the filename in the constructor.

In general you don't want to couple utility classes like this so tightly to the rest of your FAT test (and if you do, you must document it).